stdlib/io/src/main/java/org/ballerinalang/stdlib/io/utils/PropertyUtils.java

```diff
@@ -0,0 +1,91 @@
+import org.ballerinalang.jvm.types.BMapType;
+import org.ballerinalang.jvm.types.BTypes;
+import org.ballerinalang.jvm.values.MapValue;
+import org.ballerinalang.jvm.values.MapValueImpl;
+import org.ballerinalang.jvm.values.MappingInitialValueEntry;
+import org.ballerinalang.jvm.values.api.BMap;
+import org.ballerinalang.jvm.values.api.BString;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.util.Enumeration;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * The utility class used to execute read/write operations to a property file.
+ *
+ */
+public class PropertyUtils {
+    private static final Logger log = LoggerFactory.getLogger(PropertyUtils.class);
+    private static final BMapType mapType = new BMapType(BTypes.typeString);
+    private static Properties properties = null;
+
+    // Read a property related to a given key and return the BString value.
+    public static BString readProperty(Reader reader, BString key) throws IOException {
+        if (properties == null) {
+            properties = new Properties();
+            properties.load(reader);
+        }
+        return org.ballerinalang.jvm.StringUtils.fromString(properties.getProperty(key.getValue()));
+    }
+
+    // Read all properties and return as a ballerina map
+    public static MapValue<BString, BString> readAllProperties(Reader reader) throws IOException {
+        if (properties == null) {
+            properties = new Properties();
+            properties.load(reader);
+        }
+        MappingInitialValueEntry.KeyValueEntry[] keyValues = new MappingInitialValueEntry
+                .KeyValueEntry[properties.stringPropertyNames().size()];
+        int i = 0;
+        for (Enumeration<?> e = properties.propertyNames(); e.hasMoreElements(); ) {
+            String key = (String) e.nextElement();
+            String value = properties.getProperty(key);
+            MappingInitialValueEntry.KeyValueEntry keyValue = new MappingInitialValueEntry.KeyValueEntry(
+                    StringUtils.fromString(key), StringUtils.fromString(value));
+            keyValues[i] = keyValue;
+            i++;
+        }
+        return new MapValueImpl<>(mapType, keyValues);
+    }
+
+    // Generate writable string content to a property file using a given map.
+    public static String getWritablePropertyContent(BMap<BString, BString> propertiesMap) {
+        Set<Map.Entry<BString, BString>> propertiesSet = propertiesMap.entrySet();
+        Iterator<Map.Entry<BString, BString>> iterator = propertiesSet.iterator();
+        String content = "";
+        while (iterator.hasNext()) {
+            Map.Entry<BString, BString> entry = iterator.next();
+            content = content +
+                   entry.getKey().getValue() + "=" + entry.getValue().getValue() + "\n";
+        }
+        log.info(content);
```

Comment:
Remove log

Reply:
Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/24978/commits/775ccc6c85284a0598f87edd9608654ee21a3bec